art/attacks/inference/membership_inference/label_only_boundary_distance.py

```diff
@@ -196,6 +202,62 @@ def calibrate_distance_threshold(
 
         self.distance_threshold_tau = distance_threshold_tau
 
+    def calibrate_distance_threshold_unsupervised(self, top_t: int, num_samples: int, max_queries: int, **kwargs):
+        """
+        Calibrate distance threshold on randomly generated samples, choosing the top-t percentile of the noise needed
+        to change the classifier's initial prediction.
+
+        | Paper link: https://arxiv.org/abs/2007.15528
+
+        :param top_t: Top-t percentile.
+        :param num_samples: Number of random samples to generate.
+        :param max_queries: Maximum number of queries.
+        :Keyword Arguments for HopSkipJump:
+            * *norm*: Order of the norm. Possible values: "inf", np.inf or 2.
+            * *max_iter*: Maximum number of iterations.
+            * *max_eval*: Maximum number of evaluations for estimating gradient.
+            * *init_eval*: Initial number of evaluations for estimating gradient.
+            * *init_size*: Maximum number of trials for initial generation of adversarial examples.
+            * *verbose*: Show progress bars.
+        """
+        from art.attacks.evasion.hop_skip_jump import HopSkipJump
+
+        x_min, x_max = self.estimator.clip_values
+
+        x_rand = np.random.rand(*(num_samples,) + self.estimator.input_shape).astype(np.float32)
+        x_rand *= x_max - x_min  # scale
+        x_rand += x_min  # shift
+
+        y_rand = np.random.randint(0, self.estimator.nb_classes, num_samples)
```

Comment:
I found a mistake - `y_rand` should be the predicted label given `x_rand` (according to Algorithm 3 of the paper). Should be something like this:
```python
y_rand = self.estimator.predict(x=x_rand)
```

Reply:
Hi @minaremeli Are you planning to add an additional commit for this item?

Reply:
Hi @beat-buesser sorry for the delay - yes I plan to add an additional commit as soon as possible (this week).

Reply:
Great, no worries!